records/generator.go

```diff
@@ -52,15 +51,16 @@ func (rg *RecordGenerator) ParseState(leader string, c Config) error {
 		hostSpec = labels.RFC952
 	}
 
-	return rg.InsertState(sj, c.Domain, c.SOARname, c.Listener, c.Masters, c.IPSources, hostSpec)
+	return rg.InsertState(sj, c.Domain, c.SOARname, c.Listener, masters, c.IPSources, hostSpec)
 }
 
 // Tries each master and looks for the leader
 // if no leader responds it errors
-func (rg *RecordGenerator) findMaster(leader string, masters []string) (state.State, error) {
+func (rg *RecordGenerator) findMaster(masters ...string) (state.State, error) {
 	var sj state.State
 
-	// Check if ZK master is correct
+	// Check if ZK leader is correct
+	leader := masters[0]
```

Comment:
it will, indeed and is already fixed on my local working copy, thanks for recognising!